controllers/user.js

```diff
@@ -0,0 +1,52 @@
+const db = require("../config/database");
+const mongoose = require("mongoose");
+const filter = require("./utils/filterDecisionToUserArray").filter;
+/*
+Expecting: req to contain the user ID that we are updating
+Expecting: req to contain req.body.eventid --> the id that the user made a decision on
+Expecting: req to contain whether they liked, disliked, or blacklisted the event in the form of a decision 
+i.e. req.body.decision = blacklist, req.body.decision = liked, req.body.decision = disliked,  
+*/
+const updateUserEventPreferences = async (req, res) => {
+  try {
+    if (!mongoose.isValidObjectId(req.body.eventID))
+      throw "Valid Event ID not passed";
+
+    /*
+    Assuming the route is going to look sometihng like "/user/:id/updateUserEventPreferences"
+    */
+    //added this util fcn just in case what we get does not match what we expect.
+
+    //$addToSet works on arrays, and if the user decision didnt exist in our db it would create it.
+    const filteredDecision = filter(req.body.decision);
+
+    const toBeUpdated = filteredDecision
```

Comment:
i'm not too familiar with the syntax in this part but it seems right to me (just need to test them all later) - rest seems good to merge

Reply:
so, im gonna merge